src/trackers/UNIT3D/bitporn.py

```
@@ -0,0 +1,178 @@
+        if category:
+            return {"category_id": self.category_ids.get(category, self.category_ids["Uncategorized"])}
+
+        basename = str(meta.basename_no_ext or "")
+        normalized_basename = re.sub(r"[^a-z0-9]+", " ", basename.casefold())
+        for category_name, pattern in self._category_patterns:
+            if re.search(pattern, normalized_basename):
+                return {"category_id": self.category_ids[category_name]}
+
+        return {"category_id": self.category_ids["Uncategorized"]}
+
+    async def get_type_id(self, meta: Meta, type: str | None = None, reverse: bool = False, mapping_only: bool = False) -> dict[str, str]:
```

Ruff (0.16.1)

[error] Function argument `type` is shadowing a Python builtin

(A002)